lib/rucio/api/replica.py

```diff
@@ -118,13 +147,28 @@ def list_replicas(dids, schemes=None, unavailable=False, request_id=None,
     if permission.has_permission(issuer=issuer, action='get_signed_url', kwargs={}):
         sign_urls = True
 
-    return replica.list_replicas(dids=dids, schemes=schemes, unavailable=unavailable,
-                                 request_id=request_id,
-                                 ignore_availability=ignore_availability,
-                                 all_states=all_states, rse_expression=rse_expression,
-                                 client_location=client_location, domain=domain,
-                                 sign_urls=sign_urls, signature_lifetime=signature_lifetime,
-                                 resolve_archives=resolve_archives, resolve_parents=resolve_parents)
+    replicas = replica.list_replicas(dids=dids, schemes=schemes, unavailable=unavailable,
+                                     request_id=request_id,
+                                     ignore_availability=ignore_availability,
+                                     all_states=all_states, rse_expression=rse_expression,
+                                     client_location=client_location, domain=domain,
+                                     sign_urls=sign_urls, signature_lifetime=signature_lifetime,
+                                     resolve_archives=resolve_archives, resolve_parents=resolve_parents)
+
+    for rep in replicas:
```

Comment:
does this overwrite the field? the existing list_replicas response should not be modified (stuff can be added though). i'm not really following the code below

Reply:
Yes, "list_replicas" returns a dict with stucture dependant on keys which were the rse name (replicas['rses']['MOCK'], replicas['rses']['MOCK2']...). 
I changed it to rse_id so that if any other part of core needs to rely on it its still ids. 
Its then the job in api to translate those keys.
So it takes the result and replaces these keys by creating a new entry for 'rses' and 'states'. 

Why shouldn't it be modified?

Reply:
This is the API here, there are clients that rely on the RSE being the RSE name.

Reply:
Do those clients use core? The rse is id in core, then before it returns from api is convereted to name

Reply:
Ah, no, if the reply of the RSE stays the same that's fine. Be careful though that the lists in here are all sorted, so rewriting them must not change the order.

Reply:
I hadn't spotted that they're sorted. This bit won't change the order but due to changes in the core the rses will be sorted by rse_id. But I thought dicts were unordered anyway?